staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -73,6 +79,25 @@ func (a *Allocator) ClaimsToAllocate() []*resourceapi.ResourceClaim {
 	return a.claimsToAllocate
 }
 
+func (a *Allocator) compileCELExpression(expression string) cel.CompilationResult {
```

Comment:
Agreed, that's better. I think a `GetOrCompile` is fine for a cache and it fits nicely into the `cel` package, so I moved it there (`structured.CELCache` -> `cel.Cache`).

I just pushed the updated branch. Please take another look, I'd like to get this merged as soon as possible because it (indirectly, through potential code conflicts) blocks [beta promotion](https://github.com/kubernetes/kubernetes/pull/127511). It's not strictly required for beta, but would be very nice to have.